### ISSUE TITLE
Optimize fitness value

### ISSUE THREAD

**Issue by germanjke** [deepmind/deepmind-research#190]
In original BKRGA paper, and in your REGAL paper we have some fitness value function which one we want to minimize. 

In original device placement task we want to minimize runtime of inference/training of deep neural network. But this value is linear and we can't apply gradient descent to this problem. 

How do you think, can we apply some loss function to update our fitness value function, e.g. square some things, maybe add some bias value? 

For example, `fitness_value = runtime**2 + some_feature` 

I have a question, REGAL have no conflicts with modify fitness value functions?
